pkg/cloud/services/gc/loadbalancer.go

```diff
@@ -128,3 +130,150 @@ func (s *Service) deleteTargetGroup(ctx context.Context, targetGroupARN string)
 
 	return nil
 }
+
+func (s *Service) describeTargetgroups() ([]string, error) {
+	groups, err := s.elbv2Client.DescribeTargetGroups(&elbv2.DescribeTargetGroupsInput{})
+	if err != nil {
+		return nil, err
+	}
+
+	targetGroups := []string{}
+	for _, group := range groups.TargetGroups {
+		if err != nil {
+			return nil, err
+		}
+		targetGroups = append(targetGroups, *group.TargetGroupArn)
+	}
+	return targetGroups, nil
+}
+
+// getAllLoadBalancers gets all elastic LBs.
+func (s *Service) getAllLoadBalancers() ([]string, error) {
+	var names []string
+	err := s.elbClient.DescribeLoadBalancersPages(&elb.DescribeLoadBalancersInput{}, func(r *elb.DescribeLoadBalancersOutput, last bool) bool {
+		for _, lb := range r.LoadBalancerDescriptions {
+			names = append(names, *lb.LoadBalancerName)
+		}
+		return true
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return names, nil
+}
+
+// getAllLoadBalancersV2 gets all network and application LBs.
+func (s *Service) getAllLoadBalancersV2() ([]string, error) {
+	var arns []string
+	err := s.elbv2Client.DescribeLoadBalancersPages(&elbv2.DescribeLoadBalancersInput{}, func(r *elbv2.DescribeLoadBalancersOutput, last bool) bool {
+		for _, lb := range r.LoadBalancers {
+			arns = append(arns, *lb.LoadBalancerArn)
+		}
+		return true
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return arns, nil
+}
+
+// getProviderOwnedLoadBalancers gets cloud provider created LB(ELB) for this cluster, filtering by tag: kubernetes.io/cluster/<cluster-name>:owned.
+func (s *Service) getProviderOwnedLoadBalancers() ([]*AWSResource, error) {
```

Comment:
nit: `getProviderOwnedLoadBalancers` and `getProviderOwnedLoadBalancersV2` looks like almost duplicated functions, can we combine these two with the use of switch statements?